shelley-mc/formal-spec/transactions.tex

```diff
@@ -1,106 +1,27 @@
-
-  \item $\AssetID$ : for each currency ID $cid$, in the MC model, we can associate
-  some tokens with this currency. A term $t\in\AssetID$ is an identifier of a
-  kind of token in a given currency $cid$. Within the currency $cid$,
-  tokens of the same type (with the same identifier $t\in\AssetID$) are fungible
-  with each other. That is, they are effectively the "same" token, in the same way that
-  every Ada coin is indistinguishable from another Ada coin. Ada
-  has the currency ID $\mathsf{adaID}$. The statement \textit{all of the tokens
-  of this currency must
-  be the same}, in a MC scheme, means that there is only one kind of term of
-  type $\AssetID$ associated
-  with $\mathsf{adaID}$, and we call it $\mathsf{adaToken}$.
-  See below for a more detailed discussion of Ada representation.
-
-  Note that currencies with different IDs may have tokens with the same names.
-  For example, a new currency $\mathsf{NotAda}$ can also have an $\mathsf{adaToken}$.
-  These are not the same tokens and are not fungible with each other.
-
-  \item $\Quantity$ is an integer type representing an amount. This type is used
-  specifically to represent an amount of a specific $\AssetID$. We associate
-  a term $q\in\Quantity$ with a specific token to keep track of how much of
-  this token there is in a given value.
-
-  \item $\Value$ is the multicurrency type used to represent
-  all tokens, including Ada. The key of this finite map type is
-  the hash of the monetary policy script for this currency.
-  A value $v\in\Value$ is a finite map that assciates to each of its
-  currency ID keys a finite map $tkns\in\AssetID \mapsto \Quantity$,
-  which represents all the tokens (of that particular
-  currency) that are contained in $v$.
-
-  The map $tkns$ associates to each key $t\in\AssetID$ the quantity of tokens of
-  this kind (i.e. with this identifier). Tokens are fungible with each other if
-  they belong to the same $\PolicyID$ and $\AssetID$.
-
-  \item $\TxOut$ : The type of outputs carried by a transaction. This type
-  is different from the Shelley $\TxOut$ type because it contains a $\Value$
-  instead of a $\Coin$
-
-  \item $\UTxOOut$ is the type of UTxO entry that gets created when a transaction
-  output is processed. In this specification, it has the same structure as
-  the transaction output $\TxOut$, but we give it a different name, taking into
-  account that $\Value$ is stored differently in the outputs of $\UTxO$ and $\Tx$
-  (due to optimization in the $\UTxO$)
-
-  \item $\UTxO$ entries are stored in the finite map $\TxIn\mapsto \UTxOOut$
-  (this type is also different than the Shelley $\UTxO$ type)
-
-\end{itemize}
-
-
-\begin{figure*}[htb]
+\begin{figure*}[t!]
   %
-  \emph{Scripts}
+  \emph{Abstract Types}
   %
   \begin{equation*}
-    \begin{array}{r@{~\in~}l@{\qquad=\qquad}lr}
-
-    \end{array}
+    \begin{array}{r@{~\in~}l@{~}lr}
+      \var{s_{mc}} & \ScriptMPS & \text{monetary policy script}
+     \end{array}
```

Comment:
- probably it does not make sense to include the appendix in the document (keep it as a tex file in the repo though), or define this type at all. Instead of `evalMPSScript` , i think it is better to use the MSig evaluator everywhere

Make a note somewhere about scripts that says something along the lines of
- any kind of script that the ledger is able to process can always be used for any purpose. In this spec, there is only one type of script MSig, but when more scripts are introduced in the future, they can all be used as forging scripts